pylutron/__init__.py

```diff
@@ -324,12 +325,25 @@ def send(self, op, cmd, integration_id, *args):
     self._conn.send(op + out_cmd)
 
   def load_xml_db(self):
-    """Load the Lutron database from the server."""
+    """Load the Lutron database from the server. If a locally cached copy is
+    available, use that instead."""
+
+    loaded_from_cache = False
+
+    if self._cache_file:
+      try:
+        with open(self._cache_file, 'rb') as f:
```

Comment:
just because a cache file exists, doesn't mean it's up to date. We should only be caching databases that haven't changed.

There's a way to query the export time from the server, which we should then compare with the db on disk, and it's older, query the new db. We shouldn't blindly assume that the local one is fresh.

Reply:
Ah, that's great; I didn't know that. How do you do it? The repeater's HTTP server doesn't support HEAD requests (I'd tried that initially). Happy to update this to check the timestamp; obviously that's far preferable.